src/edu/csus/ecs/pc2/core/IInternalController.java

```diff
@@ -854,8 +854,9 @@ void submitJudgeRun(Problem problem, Language language, SerializedFile mainFile,
      * @param additionalFiles
      * @param overrideTimeMS
      * @param overrideRunId
+     * @throws Exception if an error occurs while attempting to send the Run to the Server
      */
-    void submitRun(ClientId submitter, Problem problem, Language language, SerializedFile mainSubmissionFile, SerializedFile[] additionalFiles, long overrideTimeMS, long overrideRunId);
+    void submitRun(ClientId submitter, Problem problem, Language language, SerializedFile mainSubmissionFile, SerializedFile[] additionalFiles, long overrideTimeMS, long overrideRunId) throws Exception;
```

Comment:
It seems like this should throw `core.exception.SubmissionRejectedException` rather than the more generic `Exception`.

Reply:
I just did what eclipse suggested.  I guess if an other exception occurs the submissions was not accepted anyway.